source/extensions/common/wasm/wasm.cc

```diff
@@ -2211,6 +2260,16 @@ void Wasm::registerCallbacks() {
 
   _REGISTER_PROXY(setEffectiveContext);
 #undef _REGISTER_PROXY
+
+#define _REGISTER_GOLANG_PROXY(_fn)                                                                  \
```

Comment:
I'm confused by why you need those? C++, Rust and TinyGo should be all calling the same functions.

Reply:
tinygo imports function with package name as prefix (`envoywasmsdk`) and two extra params appended at the end: https://tinygo.org/compiler-internals/calling-convention/.

Reply:
This is wat of tinygo generated wasm file:

``` 
(import "env" "envoywasmsdk._proxy_getHeaderMapValue" (func $envoywasmsdk._proxy_getHeaderMapValue (type $t2)))
  (import "env" "envoywasmsdk._proxy_log" (func $envoywasmsdk._proxy_log (type $t3)))
  (import "env" "envoywasmsdk._proxy_addHeaderMapValue" (func $envoywasmsdk._proxy_addHeaderMapValue (type $t2)))
```

Reply:
https://github.com/tinygo-org/tinygo/pull/455

Reply:
I just came across this PR. Please note that the calling convention should be considered an implementation detail. If you want a stable calling convention, use `//export` which is intended to match the C calling convention.